File: LeetCode Problems/901 to 1000/921. Minimum Add to Make Parentheses Valid/921. Minimum Add to Make Parentheses Valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<char> parentheses;
        int cnt = 0;
        for(char ch : s) {
            if(ch == '(')
                parentheses.push(ch);
            else {
                if (!parentheses.empty())
                    parentheses.pop();
                else
                    cnt++;                
            }
        }
        return cnt + parentheses.size();
    }
};

File: LeetCode Problems/901 to 1000/921. Minimum Add to Make Parentheses Valid/921. Minimum Add to Make Parentheses Valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int cnt = 0;
        String str = "";
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(')
                str += s.charAt(i);
            else {
                if (s.charAt(i) == ')' && str.length() > 0) {
                    StringBuilder sb = new StringBuilder(str);
                    sb.deleteCharAt(0);
                    str = sb.toString();
                } else
                    cnt++;
            }
        }
        return cnt + str.length();
    }
}
